app/forms/statement_of_cases/statement_of_case_form.rb

```diff
@@ -26,69 +26,85 @@ def self.max_file_size
       MAX_FILE_SIZE
     end
 
+    def exclude_from_model
+      [:upload_button_pressed]
+    end
+
     validates :statement, presence: true, unless: :file_present_or_draft?
-    validate :original_file_too_big
-    validate :original_file_empty
-    validate :original_file_malware_scan
-    validate :original_file_disallowed_content_type
+    validate :original_files_valid
 
     private
 
+    def original_files_valid
+      if @upload_button_pressed == true
+        if original_files.nil? || original_files.none?
+          errors.add(:original_files, original_file_error_for(:no_file_chosen))
+          return
+        end
+      end
+      original_files&.each do |original_file|
+        original_file_too_big(original_file)
+        original_file_empty(original_file)
+        original_file_malware_scan(original_file)
+        original_file_disallowed_content_type(original_file)
+      end
+    end
+
     def file_present_or_draft?
-      file_present? || draft?
+      model.original_files.any? || original_files.any? || draft?
```

Comment:
this fails if the user submits the form without any file but only the statement text.
this works, however:
`model.original_files&.any? || original_files&.any? || draft?`

I was expecting this to be picked up by the spec's context `'file is empty but text is entered'`. (line 81 of the spec file)
And I think this is because the test has params like : 
```ruby
statement_of_case: {
  statement: entered_text,
  original_files: [nil]
}
```

However, for this scenario, the params sent by the form are:
```ruby
statement_of_case: {
  statement: entered_text
}
```

Could you modify the test?

Reply:
Actually I've rewritten the specs - because there are three scenarios now: upload button pressed, continue button pressed, and save as draft button pressed.